xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -1171,6 +1205,11 @@ struct AppEnvironment {
     var documentURL: URL
     var applicationSupportURL: URL
 
+    private var _noosphere: NoosphereService
+    var noosphere: some NoosphereServiceProtocol { _noosphere }
+    /// Exposes sphere publisher protocol for Noosphere
+    var sphere: some SpherePublisherProtocol { _noosphere }
```

Comment:
We use `some` to expose only certain protocols from the caller side.

`some` does caller-side type-erasure, so that the type remains concrete in `AppEnvironement`, but the caller may only interact with the protocol, and not the underlying type.

The protocols we expose are "UI-thread-friendly". That is, if they are blocking, they are very fast, and if they aren't very fast, they produce Combine publishers and run on a separate thread.